Makefile

```diff
@@ -72,6 +72,12 @@ coverage: test
 	cat $(COVERAGE_FILE) | grep -v "_mock.go" > $(COVERAGE_FILE).no-mocks
 	go tool cover -func=$(COVERAGE_FILE).no-mocks
 
+validate-modules:
+	@echo "- Verifying that the dependencies have expected content..."
+	go mod verify
+	@echo "- Checking for any unused/missing packages in go.mod..."
+	go mod tidy
```

Comment:
Note that we would need to run a diff or something here to ensure no differences. 
I think something like the following should work (since we're not using vendoring):
```
git diff --exit-code HEAD -- go.mod go.sum
```

Reply:
True, thanks for catching that

Reply:
Done, please check now